### PR TITLE
Added index for faster GF queries

### LINKED ISSUE #1226: factoryinsight has long running distinct queries
### Expected Behavior

Distinct queries should run quick, especially the ones used in factoryinsight to show potential values.

### Current Behavior

They take up to 55 seconds (and probably even more)

### Relevant log output

_No response_

### Steps to Reproduce

1. Have a lot of times-series data
2. Try to select on the the tags in Grafana

### Context (Environment)

Beverage customer

### Possible Solution

See also unpublished knowledgebase article: https://www.notion.so/united-manufacturing-hub/ab58fbca87834bfeb7f654af1dbb43fd?v=61b50b6953fd415bb2ac4dc5dccfc109&p=9a3e71ba952a4727b22e23dcabdbbd45&pm=s

### Version

v0.9.x

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Scarjit**
# Description

This pr partly fixes #1226 by inserting an index: `CREATE INDEX ON processvaluetable(valuename, asset_id) WITH (timescaledb.transaction_per_chunk);`.

It also introduces gzip compression for any factoryinsight response, decreasing transmitted data from ~30-70% depending on the request.

Also reverts #1193

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

- [X] Builds
- [X] Responds correctly to Grafana requests

# Checklist:

- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding changes to the changelog (if needed).
- [ ] I have made corresponding notices to the upgrade instructions (if needed)

### CODE REVIEW

**Comment by Scarjit** [comment]
CodeQL fails, due to reformatting.

File: golang/cmd/factoryinsight/http.go [hunk 1]
```diff
@@ -50,7 +55,8 @@ func SetupRestAPI(accounts gin.Accounts, version string) {
 
 func handleInternalServerError(c *gin.Context, err error) {
 
-	zap.S().Errorw("Internal server error",
```
Comment (JeremyTheocharis) [inline golang/cmd/factoryinsight/http.go:53]: Can we move all these reformatting stuff into a separate commit? it kind of blows this while PR

**Review by JeremyTheocharis — CHANGES_REQUESTED** [review]

File: golang/cmd/factoryinsight/http.go (hunk 1 above)
Comment (JeremyTheocharis) [reply golang/cmd/factoryinsight/http.go:53]: @Scarjit you are still working in this?

Comment (Scarjit) [reply golang/cmd/factoryinsight/http.go:53]: @JeremyTheocharis yes, working on this now, was waiting for my large codechange (the one with the new pre-push to be merged, cause it resolves this issue)
